### ISSUE TITLE
Fatal Exception: android.content.ActivityNotFoundException No Activity found to handle Intent { act=android.intent.action.VIEW dat=https://www.paypal.com/... }

### ISSUE THREAD

**Issue by luof** [braintree/braintree_android#799]
### Braintree SDK Version

4.36.0

### Environment

Production

### Android Version & Device

Samsung Motorola LGE Google 10/11/12/13

### Braintree dependencies

api 'com.braintreepayments.api:paypal:4.36.0'
api 'com.braintreepayments.api:data-collector:4.36.0'
api 'com.braintreepayments.api:venmo:4.36.0'
api 'com.braintreepayments.api:three-d-secure:4.36.0'

### Describe the bug

Fatal Exception: android.content.ActivityNotFoundException
No Activity found to handle Intent { act=android.intent.action.VIEW dat=https://www.paypal.com/... }

android.app.Instrumentation.checkStartActivityResult (Instrumentation.java:2114)
android.app.Instrumentation.execStartActivity (Instrumentation.java:1770)
android.app.Activity.startActivityForResult (Activity.java:5473)
androidx.activity.ComponentActivity.startActivityForResult (ComponentActivity.java:712)
android.app.Activity.startActivityForResult (Activity.java:5426)
androidx.activity.ComponentActivity.startActivityForResult (ComponentActivity.java:693)
android.app.Activity.startActivity (Activity.java:5817)
android.app.Activity.startActivity (Activity.java:5770)
com.braintreepayments.api.BrowserSwitchClient.start (BrowserSwitchClient.java:70)
com.braintreepayments.api.BraintreeClient.startBrowserSwitch (BraintreeClient.kt:307)
com.braintreepayments.api.PayPalClient.startBrowserSwitch (PayPalClient.java:303)
com.braintreepayments.api.PayPalClient$4.onResult (PayPalClient.java:271)
com.braintreepayments.api.PayPalInternalClient$1$1$1.onResult (PayPalInternalClient.java:85)
com.braintreepayments.api.HttpClient$2.run (HttpClient.java:123)
android.os.Handler.handleCallback (Handler.java:938)
android.os.Handler.dispatchMessage (Handler.java:99)
android.os.Looper.loopOnce (Looper.java:233)
android.os.Looper.loop (Looper.java:344)
android.app.ActivityThread.main (ActivityThread.java:8212)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:584)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1034)

### To reproduce

open paypal webview

### Expected behavior

no crash

**Comment by luof** [comment]
getBrainTreeClient(activity, authorizationToken) { braintreeClient ->
            payPalClient = PayPalClient(braintreeClient)
            dataCollector = DataCollector(braintreeClient)
            braintreeClient.getConfiguration { configuration, configError ->
                dataCollector?.collectDeviceData(activity) { deviceData, dataCollectorError ->
                    this.deviceData = deviceData
                }
                payPalClient?.tokenizePayPalAccount(activity,
                        createPayPalCheckoutRequest( )) { payPalError: Exception? ->
                         }
}

  <intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data android:scheme="${applicationId}.braintree" />
   </intent-filter>

**Comment by sarahkoop** [comment]
Based on the integration pattern you are using, you need to manually handle the browse switch by invoking [payPalClient.onBrowserSwitchResult](https://github.com/braintree/braintree_android/blob/main/PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java#L357C17-L357C38). Closing this issue as it appears to be an integration problem, not an issue within the SDK.

**Comment by luof** [comment]
```
        if (activity.isFinishing()) {
            String activityFinishingMessage =
                    "Unable to start browser switch while host Activity is finishing.";
            throw new BrowserSwitchException(activityFinishingMessage);
        } else if (browserSwitchInspector.deviceHasChromeCustomTabs(appContext)) {
            boolean launchAsNewTask = browserSwitchOptions.isLaunchAsNewTask();
            customTabsInternalClient.launchUrl(activity, browserSwitchUrl, launchAsNewTask);
        } else {
            Intent launchUrlInBrowser = new Intent(Intent.ACTION_VIEW, browserSwitchUrl);
            activity.startActivity(launchUrlInBrowser);  // <- Crash at here
        }
```

I called this method `tokenizePayPalAccount ` and finally it will run at `activity.startActivity(launchUrlInBrowser);` crash is happed here, before I call `onBrowserSwitchResult`, so `onBrowserSwitchResult` cant be called

**Comment by sarahkoop** [comment]
Ah ok I see the issue here - does the device you are testing on now have a web browser app installed?

**Comment by luof** [comment]
we dont know what exactly the browser installed, we just get this crash information from firebase can you fix this crash if browser not installed

**Comment by sarahkoop** [comment]
@luof yes we can return an error rather than crash if the device doesn't have a browser. However, the PayPal flow will not be possible without a web browser on the device.

**Comment by sshropshire** [comment]
@luof are you able to reproduce this error?

**Comment by sshropshire** [comment]
@luof it will be difficult to give you any approximation without additional information. It sounds like more investigation is needed on your end.

**Comment by luof** [comment]
No,I cant, I just get this crash information from firebase

**Comment by sshropshire** [comment]
@luof we'll have to do some investigation on our end before we can determine the root cause of the issue. Android devices without browser should be a rare, especially for production environments.

In the meantime as a workaround, are you able to do the check on your side to make sure the device has a browser before launching the PayPal flow?

**Comment by hollabaq86** [comment]
for internal tracking, issue 30172

**Comment by sarahkoop** [comment]
This fix was released in version `4.40.1`
